### PR TITLE
Bugfix/service endpoints include only public methods

### PR DESCRIPTION
**Pull request by anwar6953**
The first commit is the bugfix and the second commit is miscellaneous other cleanup within the test class.

https://issues.cask.co/browse/CDAP-3044
http://builds.cask.co/browse/CDAP-RBT327-4

### CODE REVIEW

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java [hunk 1]
```diff
@@ -529,6 +533,32 @@ public void testWorkerSpecification() throws Exception {
...
+                                                                 new ServiceHttpEndpoint("GET", "/multi"),
+                                                                 new ServiceHttpEndpoint("GET", "/multi/ping"));
+
+    GsonBuilder gsonBuidler = new GsonBuilder();
+    gsonBuidler.registerTypeAdapter(ServiceSpecification.class, new ServiceSpecificationCodec());
+    gsonBuidler.registerTypeAdapter(HttpServiceHandlerSpecification.class, new HttpServiceSpecificationCodec());
+    Gson gson = gsonBuidler.create();
+    ServiceSpecification specification = readResponse(response, ServiceSpecification.class, gson);
+
+    Set<ServiceHttpEndpoint> returnedEndpoints = Sets.newHashSet();
```
Comment (bdmogal) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: use Java 7 syntax?

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java [hunk 2]
```diff
@@ -851,9 +877,8 @@ private HttpResponse sendLiveInfoRequest(String namespace, String appId, String
     return doGet(versionedUrl);
   }
 
-  private int changeFLowletStreamConnection(String namespace, String appId, String flow, String flowlet,
-                                            String oldStreamId, String newStreamId)
-    throws Exception {
+  private int changeFlowletStreamConnection(String namespace, String appId, String flow, String flowlet,
```
Comment (bdmogal) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: should we have this method accept ``Id.Flow`` too?

**Comment by bdmogal** [comment]
Nice cleanup! just a couple of minor comments, otherwise :+1: !

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java (hunk 2 above)
Comment (anwar6953) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: this method wasn't being used, so I just left it as is

**Comment by anwar6953** [comment]
@bdmogal Addressed comments.

**Comment by bdmogal** [comment]
:+1:

**Pull request merged**